researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -540,17 +540,25 @@ public void setUserSharingScope(Context context, String scope) {
         return bridgeManagerProvider.getParticipantManager()
                 .updateParticipantRecord((StudyParticipant) new StudyParticipant()
                         .email(authenticationManager.getEmail())
-                        .sharingScope(scope));
+                        .sharingScope(scope))
+                .doOnSuccess(session -> bridgeManagerProvider.getAccountDao()
```

Comment:
All these setDataGroups spots seem to stem from the user session, maybe the call to "setUserSession" should go through and update the study participant, data groups, and session?